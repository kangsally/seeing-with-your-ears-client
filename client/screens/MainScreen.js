import React, { Component } from 'react';
import { View, StyleSheet, Image } from 'react-native';
import MainButtons from '../components/MainButtons';
import InstructionBar from '../components/InstructionBar';
import { startTospeak, stopToSpeak } from '../utils';
import { MAIN_TITLE } from '../constants/titles';
import { CAMERA_SCREEN, LOCATION_SCREEN } from '../constants/screens';
import { mainScreenStartScript } from '../assets/audioScripts';

export default class MainScreen extends Component {
  componentDidMount = () => {
    startTospeak(mainScreenStartScript);
  };

  navigateBtn = navigate => {
    const { navigation } = this.props;
    stopToSpeak();
    if (navigate === 'right') {
      navigation.navigate(CAMERA_SCREEN);
    } else if (navigate === 'left') {
      navigation.navigate(LOCATION_SCREEN);
    }
  };

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.content}>
          <View style={styles.imageBox}>
            <Image
              style={styles.image}
              source={require('../assets/logo-1.png')}
            />
          </View>
          <InstructionBar content={MAIN_TITLE} />
        </View>
        <MainButtons onPressBtn={this.navigateBtn} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  content: {
    flex: 2,
    backgroundColor: '#fff'
  },
  image: {
    justifyContent: 'center',
    alignItems: 'center',
    width: '50%',
    height: '60%'
  },
  imageBox: {
    flex: 8,
    justifyContent: 'center',
    alignItems: 'center'
  }
});
