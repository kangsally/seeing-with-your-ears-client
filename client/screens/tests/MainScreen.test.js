import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MainScreen from '../MainScreen';
Enzyme.configure({ adapter: new Adapter() });

describe('MainScreen snapshot', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<MainScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('MainScreen components', () => {
  const wrapper = shallow(<MainScreen />);
  it('should render a MainButton component', () => {
    expect(wrapper.find('MainButtons')).toHaveLength(1);
  });
  
  it('should render a InstructionBar component', () => {
    expect(wrapper.find('InstructionBar')).toHaveLength(1);
  });

  it('should render a View component', () => {
    expect(wrapper.find('View')).toHaveLength(3);
  });
});
