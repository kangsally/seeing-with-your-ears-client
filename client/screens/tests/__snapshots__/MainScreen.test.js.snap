// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen snapshot matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 8,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "alignItems": "center",
            "height": "60%",
            "justifyContent": "center",
            "width": "50%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1A1A1A",
          "flex": 2,
          "justifyContent": "center",
          "marginTop": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        귀로 메인 화면
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessibilityLabel="left"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#EAEDEC",
          "borderRightColor": "#868e96",
          "borderRightWidth": 0.5,
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      accessibilityLabel="right"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#EAEDEC",
          "borderLeftColor": "#868e96",
          "borderLeftWidth": 0.5,
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
